### ISSUE TITLE
Label LineHeight does not render the same on iOS and Android

### ISSUE THREAD

**Issue by adrianknight89** [xamarin/Xamarin.Forms#6046]
### Description

If you change the default `LineHeight` on a `Label`, when text is wrapped, Android seems to be applying spacing between lines. On the other hand, iOS changes the height of each line, which creates unnecessary padding above the first line.

### Steps to Reproduce
Run the attached repro on both platforms and notice that iOS line spacing is off.

### Reproduction Link

[App18776.zip](https://github.com/xamarin/Xamarin.Forms/files/3129362/App18776.zip)

**Comment by kingces95** [comment]
Android on master:
![image](https://user-images.githubusercontent.com/4120386/57043732-251f7b80-6c04-11e9-9806-a8c462bc50e2.png)

**Comment by kingces95** [comment]
iOS on master
![Image from iOS](https://user-images.githubusercontent.com/4120386/57051784-e5b25880-6c1e-11e9-899a-21c3f83403e1.png)

**Comment by kingces95** [comment]
So I see a space above the first line on iOS. @adrianknight89  What's the issue on android? That it's just different than iOS or is there something more?

**Comment by adrianknight89** [comment]
I don't believe anything is wrong with Android. This seems to be an iOS-only issue.

**Comment by misjavs** [comment]
Hi,

I'm also experiencing this issue, and it looks as if the wrong property is used in the iOS renderer:
![image](https://user-images.githubusercontent.com/13250852/71666126-eb3ce100-2d5f-11ea-939f-910ecec747bc.png)

This could be a solution:
#`https://github.com/xamarin/Xamarin.Forms/compare/master...misjavs:patch-1`

I haven't tested it..

**Comment by MitchBomcanhao** [comment]
if the Xamarin property is line height, then it should be changing the entire line height, not just the spacing between lines. I'd argue that it is the android implementation that is buggy, not iOS.

**Comment by misjavs** [comment]
I agree.
However, the LineHeight property is unusable at the moment because of the platform differences and the inability to perfectly center a text in the middle. 
The LineSpacing property seems like a much easier to use property in various UI's,

**Comment by malware-dev** [comment]
We also have a big problem with this. Both the fact that nothing happens with a single line on Android, _as well_ as the fact that the lines on iOS is aligned to the bottom of each individual line.

We are dependant on text rendering the same way on both platforms.

**Comment by LiamMartens** [comment]
Any decent workarounds for this? The android implementation seems nonsensical at this time

**Comment by jmielczarkowski** [comment]
@LiamMartens @adrianknight89 

on both platforms i's working similar, it's just line spacing multiplier 
1. create a sample page
2. put some labels with line-height from 1 to 1.5 
3. run on Android / iOS, see both line heights are in different sizes

**Comment by Elifas-TeQ** [comment]
Hello there, I have a trouble with this as well. Actually, I agree with @MitchBomcanhao rather than with the author of the issue.
`LineHeight` should change height of each line of text for a label. Basing on that, I would say that iOS works as expected but Android doesn't.
@jmielczarkowski I tried. Here is the result.
So, simple code:
`<StackLayout BackgroundColor="LightYellow">
    <Label
        LineHeight="2"
        VerticalOptions="CenterAndExpand"
        VerticalTextAlignment="Center"
        BackgroundColor="Orange"
        Text="..." /></StackLayout>`
_Once again, iOS works perfectly, from my point of view._
There are two problems for Android though:
1. If there is only one line of text then `LineHeight` does not affect height **at all**:
![Screenshot 2022-03-15 at 17 23 18](https://user-images.githubusercontent.com/21258765/158416556-96174971-57c9-46bd-8f6f-f02f3def71b5.png)
2. If more - `LineHeight` changes spacing between lines, not actual height of line:
![Screenshot 2022-03-15 at 17 26 58](https://user-images.githubusercontent.com/21258765/158416559-98a9d8a7-c2cb-4e31-b5ae-2a56b398c6b7.png)
![Screenshot 2022-03-15 at 17 27 08](https://user-images.githubusercontent.com/21258765/158416562-8339aed4-d6a5-4db1-8178-77b04d4c14ab.png)
![Screenshot 2022-03-15 at 17 27 35](https://user-images.githubusercontent.com/21258765/158416564-e682b412-7229-4314-8dae-0578b1b79900.png)
So, unfortunately, I see the problem here. It does not work in the same way for different platforms. And it does not work as expected.
Could you, please, take a look on it?
Regards, Oleksandr

**Comment by daniel-streetec** [comment]
Hi, were you able to resolve this problem? I have the same problem using the LineHeight and VerticalTextAlignment properties

![IMG_0008](https://user-images.githubusercontent.com/43196111/204994059-9b51e977-cff1-4669-8c50-16eed4575c47.PNG)
